core/misc/slot_chain.go

```diff
@@ -0,0 +1,94 @@
+package misc
+
+import (
+	"sync"
+
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/alibaba/sentinel-golang/core/log"
+	"github.com/alibaba/sentinel-golang/core/stat"
+)
+
+var (
+	rsSlotChainLock sync.RWMutex
+	rsSlotChain = make(map[string]*base.SlotChain, 8)
+)
+
+func validateRuleCheckSlot(sc *base.SlotChain, s base.RuleCheckSlot) bool {
+	flag := false
+	f := func(slot base.RuleCheckSlot) {
+		if slot.Name() == s.Name() {
+			flag = true
+		}
+	}
+	sc.RangeRuleCheckSlot(f)
+
+	return flag
+}
+
+func newSlotChain() *base.SlotChain {
+	sc := base.NewSlotChain()
+	sc.AddStatPrepareSlotLast(stat.DefaultResourceNodePrepareSlot)
+
+	sc.AddStatSlotLast(stat.DefaultSlot)
+	sc.AddStatSlotLast(log.DefaultSlot)
+
+	return sc
+}
+
```

Comment:
Maybe we also need to provide API `RegisterResourceStatPrepareSlot`?

Reply:
In fact there is only one prepare slot now. So I do not think we need this API at present.